src/zkevm_specs/evm_circuit/execution/error_gas_uint_overflow.py

```diff
@@ -0,0 +1,176 @@
+from zkevm_specs.evm_circuit.util.call_gadget import CallGadget
+from ...util import (
+    FQ,
+    TxDataNonZeroGasEIP2028,
+    MAX_U64,
+    N_BYTES_U64,
+    TxGas,
+    TxGasContractCreation,
+    TxDataZeroGas,
+)
+from ..instruction import Instruction, Transition
+from ..table import CallContextFieldTag
+from ..opcode import Opcode
+
+
+def error_gas_uint_overflow(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    (
+        is_call,
+        is_callcode,
+        is_delegatecall,
+        is_staticcall,
+        is_create_flag,
+        is_create2_flag,
+        is_calldatacopy,
+        is_codecopy,
+        is_extcodecopy,
+        is_returndatacopy,
+        is_log0,
+        is_log1,
+        is_log2,
+        is_log3,
+        is_log4,
+        is_sha3,
+        is_exp,
+        is_mload,
+        is_mstore,
+        is_mstore8,
+        is_return,
+        is_revert,
+    ) = instruction.multiple_select(
+        opcode,
+        (
+            Opcode.CALL,
+            Opcode.CALLCODE,
+            Opcode.DELEGATECALL,
+            Opcode.STATICCALL,
+            Opcode.CREATE,
+            Opcode.CREATE2,
+            Opcode.CALLDATACOPY,
+            Opcode.CODECOPY,
+            Opcode.EXTCODECOPY,
+            Opcode.RETURNDATACOPY,
+            Opcode.LOG0,
+            Opcode.LOG1,
+            Opcode.LOG2,
+            Opcode.LOG3,
+            Opcode.LOG4,
+            Opcode.SHA3,
+            Opcode.EXP,
+            Opcode.MLOAD,
+            Opcode.MSTORE,
+            Opcode.MSTORE8,
+            Opcode.RETURN,
+            Opcode.REVERT,
+        ),
+    )
+    is_create = is_create_flag + is_create2_flag
+
+    # IntrinsicGas
+    # https://github.com/ethereum/go-ethereum/blob/b946b7a13b749c99979e312c83dce34cac8dd7b1/core/state_transition.go#L67
+    calldata_offset = instruction.call_context_lookup(CallContextFieldTag.CallDataOffset)
+    calldata_length = instruction.call_context_lookup(CallContextFieldTag.CallDataLength)
+    data = [
+        instruction.tx_calldata_lookup(tx_id, calldata_offset + FQ(idx))
+        for idx in range(calldata_length.expr().n)
+    ]
+    dataLen = len(data)
+
+    def non_zero_gas_constraints():
+        # eip 2028
+        nz = len([byte for byte in data if byte != 0])
+        gas = TxGasContractCreation if is_create == FQ(1) else TxGas
+        is_eip2028_overflow, _ = instruction.compare(
```

Comment:
I would suggest to replace `eip2028` with `non_zero_calldata`. So, the var name would be `is_non_zero_calldata_gas_overflow`. It's more self-explanatory and readers don't need to go to check what EIP-2028 is.